### ISSUE TITLE
Generated resource handler should handle security checking by default

### ISSUE THREAD

**Issue by windhood** [jhipster/generator-jhipster#1360]
for example if you generate a Book entity, in BookResource controller, the update and delete function should not allow the current user touch other user's book unless she is an admin. the getAll function should generally return current user's books unless she is an admin

**Comment by jdubois** [comment]
By default we do a simple "CRUD", so we do a "getAll".
If you want to restrict it to the current user, which is a common use case (of course, you are right!), then you can do a many-to-one relationship to the user entity (documented [here](http://jhipster.github.io/creating_an_entity.html) but it's hidden in a big page!), and that will provide you with a specific query so you can filter the data for the current user only.
So basically we provide what you ask for, but that's not the default behavior as for us that's a business decision, so we can only provide you helpers to makes it easier.

**Comment by windhood** [comment]
Thanks! I noticed that. what I want really is the controller class also can handle this default behaviour. such as one user can not update another user's book, one user can only fetch her own books unless she is an admin.  Also how do i set the #{principle.username} when unit test the findAllForCurrentUser function in repository?
